components/notebook-controller/Dockerfile

```diff
@@ -3,32 +3,60 @@
 # The Docker context is expected to be:
 #
 # ${PATH_TO_KUBEFLOW/KUBEFLOW repo}/components
-#
-# This is necessary because the Jupyter controller now depends on
-# components/common
+
+
+# Build arguments
+ARG SOURCE_CODE=.
 ARG GOLANG_VERSION=1.20
-FROM golang:${GOLANG_VERSION} as builder
 
+# Use ubi8/go-toolset as base image
+FROM registry.redhat.io/ubi8/go-toolset:${GOLANG_VERSION} as builder
+
+## Build args to be used at this step
+ARG SOURCE_CODE
```

Comment:
This is defined on line 9 already, is this intentional?

Reply:
You have to repeat it after the `FROM` line in order to be able to actually use it in that particular image build. And that's what cpaas repo does and what we release, so in this PR it's never wrong to copy from it verbatim, I think.